app/poros/monster.rb

```diff
@@ -25,5 +26,22 @@ def initialize(monster_data)
     @special_abilities = monster_data['special_abilities']
     @actions = monster_data['actions']
     @legendary_actions = monster_data['legendary_actions']
+    @proficiencies = monster_proficiencies(monster_data['proficiencies'])
+    @senses = monster_data['senses']
+    @damage_immunities = monster_data['damage_immunities']
+    @condition_immunities = monster_data['condition_immunities']
+    @damage_resistances = monster_data['damage_resistances']
+    @damage_vulnerabilities = monster_data['damage_vulnerabilities']
+    @proficiency_bonus = monster_data['proficiency_bonus']
+  end
+  
+  private
+  
+  def monster_proficiencies(raw_proficiency_data)
+    if !raw_proficiency_data.nil?
```

Comment:
My comment is largely stylistic preference, so fine as is if you don't want to change. However, I find the double negatives sometimes get a little confusing to mentally parse. I'm partial to writing in the affirmative when it makes sense. So something like:
`raw_proficiency_data.present?` instead of check for not nil

You could also take this one step further and implement a guard clause to avoid the iteration within an`if`. 
```ruby
    return unless raw_proficiency_data.present?

    raw_proficiency_data.map do |proficiency|
      { 'value' => proficiency['value'], 'name' => proficiency['proficiency']['name']}
    end
```

I ran your tests after changing to the above and they still passed. Assuming the tests are good, a change like this wouldn't break anything.

Reply:
Fantastic feedback! I will change this now.